src/figuresClasses.ts

```diff
@@ -1,19 +1,82 @@
 export interface Figure {
+  shape: string;
+  color: string;
+  getArea(): number;
+}
 
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+type TColor = 'blue' | 'red' | 'green';
+
+export class Triangle implements Figure {
+  shape: Shape.Triangle;
+
+  constructor(public color: TColor, public a: number,
+    public b: number, public c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Side lengths must be greater than 0');
+    }
+
+    if (a + b <= c || c + b <= a || c + a <= b) {
+      throw new Error('Invalid triangle sides');
+    }
+    this.shape = Shape.Triangle;
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
```

Comment:
during modern syntaxis you can remove these lines and everything will works like now